balancer_switching_test.go

```diff
@@ -131,3 +131,205 @@ func TestSwitchBalancer(t *testing.T) {
 		t.Fatalf("check pickfirst returned non-nil error: %v", err)
 	}
 }
+
+func TestSwitchBalancerGRPCLB(t *testing.T) {
+	defer leakcheck.Check(t)
+	r, rcleanup := manual.GenerateAndRegisterManualResolver()
+	defer rcleanup()
+
+	cc, err := Dial(r.Scheme()+":///test.server", WithInsecure(), WithCodec(testCodec{}))
+	if err != nil {
+		t.Fatalf("failed to dial: %v", err)
+	}
+	defer cc.Close()
+
+	r.NewAddress([]resolver.Address{{Addr: "backend"}})
+	var isPickFirst bool
+	for i := 0; i < 1000; i++ {
+		cc.mu.Lock()
+		isPickFirst = cc.curBalancerName == pickfirstName
+		cc.mu.Unlock()
+		if isPickFirst {
+			break
+		}
+		time.Sleep(time.Millisecond)
+	}
+	if !isPickFirst {
+		t.Fatalf("after 1 second, cc.balancer is of type %v, not pick_first", cc.curBalancerName)
```

Comment:
Is 1s long enough for these tests on travis?  I'd rather give them 5-10 to avoid potential flakiness -- they stop early when the condition occurs, so it's not harmful to give them longer unless the timing is important.